daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -712,6 +712,19 @@ def write_results(self, result):
                 elif self.output_parser is DropParser.NPY:
                     drop_loaders.save_npy(o, r)
                 elif self.output_parser is DropParser.RAW:
+                    if not isinstance(r, (bytes, memoryview, str)):
+                        raise Exception(
```

Comment:
I think these exceptions should be marked as [TypeErrors](https://docs.python.org/3/library/exceptions.html#TypeError).